docs/DeveloperGuide.md

```diff
@@ -1,25 +1,283 @@
+The results are stored in a `HashMap<String, String>`, which is a `HashMap` of key-value pairs, similar to the form of the
+optional argument (`<key>/<value>`). The results are returned to the `parse` method and stored as `argumentsMap`.
+1. The method then checks the `rootCommand` and decides which `Command` to return, which calls `CommandCreator`
+methods with the parsed `argumentsMap`, `description`, and `commandString`.
+1. The results of the `CommandCreator` methods are returned as a `Command` back to the invoker of the `parse` method.
+
+#### Regular expression parsing
+
+Two of the previously mentioned methods, `removeArgumentsFromCommand` and `getArgumentsFromRegex` make use of regular
+expressions to parse the optional arguments.
+
+- The regular expression that parses these optional arguments is `([\w]+/[^\s]+)`. This regular expression matches 1 or more
+alphanumeric characters (denoted by `[\w]+`), followed by a forward slash, then 1 or more of any character except whitespace (denoted by `[^\s]+`).
+- The expression also uses capturing parenthesis to ensure that the parser does not parse the same argument twice.
+
+#### Design considerations
+
+- The results of `getArgumentsFromRegex` are stored as a `HashMap` instead of `ArrayList` or simply returned as a value.
+This allows the same method to be reused for different commands, which may accept different optional arguments with
+different key-value pairs. This ensures that the code follows DRY principles.
+- The regular expressions parsing means that we do not need to manually parse every different command with different
+arguments, thus reducing code complexity and SLOC.
+
+<!-- @@author -->
+### Command component
+
+<!-- @@author GuoAi -->
+### Storage component  
+
+The `Storage` class is a class loading data from files when Duke starts and saving data to files after each command.
+
+#### High level description
+
+Methods handling data loading (i.e. `loadTask()`, `loadBook()`, `loadLinks()`, `loadCredit()`, `loadModule()` methods) 
+return an `ArrayList` of `Item`s (i.e. `Task`, `Book`, `Link`, `Credit`, `Module`). These will be the initial values of 
+the `ItemList`. The `save()` method takes an `ItemList` and a `String` specifying the path to which the file will be 
+saved. The `ItemList` will be parsed and saved into files (each `ItemList` will be saved to a separate file) at the 
+specified path.  
+  
+Formats of the files: 
+  
+- `tasks.txt`:  
+  
+There are 6 fields stored for each `Task`:  
+1. String `T` for "Task"  
+2. Whether the `Task` has been done or not (1 for done, 0 for not done)  
+3. Description of the `Task`  
+4. Priority of the `Task` (an Integer)
+5. Category of the `Task`
+6. Date of the `Task`  
+  
+All the fields are separated by ` | ` with a leading and a trailing space. Each `Task` is stored as one line.  
+  
+Example: `T | 0 | borrow book | 1 | book | 28-10-2020`  
```

Comment:
Good job! I like how this format is used consistently to explain the store files.